### ISSUE TITLE
Improvements to multivariate output emulator

### ISSUE THREAD

**Issue by jgomezdans** [jgomezdans/gp_emulator#12]
It would be advantageous to train multi-angle, spectral output emulators with the same basis functions. Basically, the PCA is done in a fairly large number of multi-angle spectra, including a significant amount of ground (LAI=0) spectra. Additionally, we could use the first two basis functions to be the soil, an then add the actual PCA components (so that we have direct access to the soil spectra).

A second thing is that for a multivariate emulator where all individual emulators have the same input training set, the distance metric is the same, so we can re-use this calculation of the K matrix for all emulators

**Comment by jgomezdans** [comment]
I guess that the distance metric calculation would require quite a lot of recalculations, as the hyperparameters are brought right in...
